xtask/src/dist.rs

```diff
@@ -58,30 +58,67 @@ fn dist_client(version: &str, release_tag: &str) -> Result<()> {
 }
 
 fn dist_server() -> Result<()> {
-    if cfg!(target_os = "linux") {
+    let target = get_target();
+    if target.contains("-linux-gnu") {
         env::set_var("CC", "clang");
     }
-    cmd!("cargo build --manifest-path ./crates/rust-analyzer/Cargo.toml --bin rust-analyzer --release").run()?;
-
-    let (src, dst) = if cfg!(target_os = "linux") {
-        ("./target/release/rust-analyzer", "./dist/rust-analyzer-linux")
-    } else if cfg!(target_os = "windows") {
-        ("./target/release/rust-analyzer.exe", "./dist/rust-analyzer-windows.exe")
-    } else if cfg!(target_os = "macos") {
-        ("./target/release/rust-analyzer", "./dist/rust-analyzer-mac")
-    } else {
-        panic!("Unsupported OS")
-    };
-
-    let src = Path::new(src);
-    let dst = Path::new(dst);
 
+    let toolchain = get_toolchain(&target);
+    cmd!("cargo +{toolchain} build --manifest-path ./crates/rust-analyzer/Cargo.toml --bin rust-analyzer --target {target} --release").run()?;
+    let src =
+        Path::new("target").join(format!("rust-analyzer{}", get_executable_extension(&target)));
+    let dst = Path::new("dist").join(get_release_artifact_name(&target));
     cp(&src, &dst)?;
     gzip(&src, &dst.with_extension("gz"))?;
 
     Ok(())
 }
 
+fn get_target() -> String {
+    match env::var("RA_TARGET") {
+        Ok(target) => target,
+        _ => {
+            if cfg!(target_os = "linux") {
+                "x86_64-unknown-linux-gnu".to_string()
+            } else if cfg!(target_os = "windows") {
+                "x86_64-pc-windows-msvc".to_string()
+            } else if cfg!(target_os = "macos") {
+                "x86_64-apple-darwin".to_string()
+            } else {
+                panic!("Unsupported OS")
+            }
+        }
+    }
+}
+
+fn get_executable_extension(target: &str) -> String {
+    if target.contains("-windows-") {
+        ".exe".into()
+    } else {
+        "".into()
+    }
+}
+
+fn get_toolchain(target: &str) -> String {
+    match target {
+        "aarch64-apple-darwin" => "beta".to_string(),
+        _ => "stable".to_string(),
+    }
+}
+
+fn get_release_artifact_name(target: &str) -> String {
```

Comment:
So the plan here is to add full target tripple to the executable name? With two different macs that makes sense actually, :+1:.

For the transition, we should adjust vscode, and broadcast the change to the clients issue. We might also just publish to sets of artifacts? THat is, run `cp rust-analyzer-windows.exe rust-analyzer-x86_64-pc-windows-msvc.exe` in `.yaml.

Reply:
> So the plan here is to add full target tripple to the executable name? With two different macs that makes sense actually, +1.

We also had requests for Windows ARM binaries, so using the target tuple seems reasonable.

Reply:
> For the transition, we should adjust vscode, and broadcast the change to the clients issue. We might also just publish to sets of artifacts? THat is, run cp rust-analyzer-windows.exe rust-analyzer-x86_64-pc-windows-msvc.exe in `.yaml.

I did it the other way around, but in `xtask`.